p2p/shim.go

```diff
@@ -0,0 +1,227 @@
+package p2p
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+var _ Reactor = (*ReactorShim)(nil)
+
+type (
+	CodecShim interface {
+		Marshal(proto.Message) ([]byte, error)
+		Unmarshal([]byte) (proto.Message, error)
+		Validate(proto.Message) error
+	}
+
+	ChannelShim struct {
+		Descriptor *ChannelDescriptor
+		Channel    *Channel
+		InCh       chan Envelope
+		OutCh      chan Envelope
+		PeerErrCh  chan PeerError
+	}
+
+	ReactorShim struct {
+		BaseReactor
+
+		Name         string
+		Codec        CodecShim
```

Comment:
We shouldn't use codecs (this was discussed during the ADR review), instead we should pass a map of channel IDs and Protobuf message types that should be used for automatic (un)marshalling.

Reply:
Yeah, I realized I could just do `proto.Unmarshal/Marshal` 👍